Simple temperature graph on main page

MaxineMuster opened the pull request:
Store the last 100 internal CPU temperatures (if present) and display them in a simple SVG:

![Graph_LN_BK_W800](https://github.com/user-attachments/assets/7f108af2-6757-43dc-9441-7ad327bcb17d)

MaxineMuster committed:
Storing and displaying CPU temperature
First try to implement drawing charts
Working on LN882H atm

MaxineMuster committed:
Enable on W800 an BEKEN, too

MaxineMuster committed:
wip - actual work

divadiow commented:
cool 
![image](https://github.com/user-attachments/assets/c06b8965-d743-4d87-b75b-7b429dd2541d)

lmk if you want it seen on any other CPU temperature supporting platforms

MaxineMuster committed:
Change to exterlan ringbuffer code

MaxineMuster committed:
Merge branch 'main' into mygraph

MaxineMuster committed:
Fix missing "bool" in ringbuff32.c for Windows and Linux

MaxineMuster committed:
try fixing bool again, fixing return in void function

MaxineMuster committed:
Add ringbuffer source file

MaxineMuster committed:
Update ringbuff32.c

MaxineMuster committed:
Update obk_main.cmake

MaxineMuster committed:
Add ringbuff32.c to obk_main.mk

MaxineMuster committed:
Merge branch 'openshwprojects:main' into mygraph

MaxineMuster committed:
Try fixing missing bool for Linux
add "fake chip temperature" for Windows

MaxineMuster committed:
Fix Windows makefile
try fixing Adress Sanitizer complaints

MaxineMuster committed:
fix missing bracket

MaxineMuster committed:
Moving all to a driver

MaxineMuster committed:
clean up changed files

MaxineMuster committed:
Merge branch 'openshwprojects:main' into mygraph

MaxineMuster committed:
save only 100 values - with 500 device became laggy after some time (understandable: 500 values translated to an JS array makes it > 1.3 k).

Added some logic only to give data in "status" refresh if data changed before.

Seems better now, but displayed time for data is drifting after some time: saved every 30 seconds, then present in next update of status.
Since I only provide the values, no timestamps, it will use actual date/time for last value and calculate the older as 30 seconds before.

But as an indicator it's o.k., I guess

MaxineMuster committed:
Enabled also for W600

MaxineMuster committed:
improve text on x axis

MaxineMuster committed:
Merge branch 'main' into mygraph

MaxineMuster committed:
for testing enable on more platforms

MaxineMuster committed:
Merge branch 'main' into mygraph

MaxineMuster committed:
Merge branch 'openshwprojects:main' into mygraph

MaxineMuster committed:
Merge branch 'openshwprojects:main' into mygraph
